File: src/main/java/com/silverboxsoft/dynamodbtool/dao/TableInfoDao.java
package com.silverboxsoft.dynamodbtool.dao;

import java.net.URISyntaxException;

import com.silverboxsoft.dynamodbtool.classes.DynamoDbConnectInfo;

import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest;
import software.amazon.awssdk.services.dynamodb.model.TableDescription;

/**
 * https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/javav2/example_code/dynamodb/src/main/java/com/example/dynamodb/DescribeTable.java
 * 
 * @author tanakaeiji
 *
 */
public class TableInfoDao extends AbsDao {

	public TableInfoDao(DynamoDbConnectInfo connInfo) {
		super(connInfo);
	}

	public TableDescription getTableDescription(String tableName) throws URISyntaxException {
		DescribeTableRequest request = DescribeTableRequest.builder()
				.tableName(tableName)
				.build();
		DynamoDbClient ddb = getDbClient();
		try {
			return ddb.describeTable(request).table();
		} finally {
			ddb.close();
		}
	}
}

File: src/main/java/com/silverboxsoft/dynamodbtool/dao/TableInfoDao.kt
package com.silverboxsoft.dynamodbtool.dao

import kotlin.Throws
import com.silverboxsoft.dynamodbtool.classes.DynamoDbConnectInfo
import java.net.URISyntaxException
import software.amazon.awssdk.services.dynamodb.model.TableDescription
import software.amazon.awssdk.services.dynamodb.model.DescribeTableRequest

/**
 * https://github.com/awsdocs/aws-doc-sdk-examples/blob/master/javav2/example_code/dynamodb/src/main/java/com/example/dynamodb/DescribeTable.java
 *
 * @author tanakaeiji
 */
class TableInfoDao(connInfo: DynamoDbConnectInfo?) : AbsDao(connInfo) {
    @Throws(URISyntaxException::class)
    fun getTableDescription(tableName: String?): TableDescription {
        val request = DescribeTableRequest.builder()
                .tableName(tableName)
                .build()
        val ddb = dbClient
        return try {
            ddb!!.describeTable(request).table()
        } finally {
            ddb!!.close()
        }
    }
}